### ISSUE TITLE
Create new helper

### ISSUE THREAD

**Issue by chulcha** [opencart/opencart#7569]
If a common/pagination controller was created, then why not create a helper

```
function pagination_result($text,$total,$page,$limt) {
return sprintf($text, ($total) ? (($page - 1) * $limit) + 1 : 0, ((($page - 1) * $limit) > ($total - $limit)) ? $total : ((($page - 1) * $limit) + $limit), $total, ceil($total / $limit));
} 
```
This will improve code readability and reduce the chance of error.
`
$data['results'] = pagination_result($this->language->get('text_pagination'),$product_total,$page,$limit);`

**Comment by diegospm** [comment]
Among other points, I sent a PR with a suggestion about this. Please see: https://github.com/opencart/opencart/pull/7490#issuecomment-505238993

Much cleaner than it is currently:

`$data['results'] = $this->load->controller('common/pagination/results', $product_total);`

Daniel ignored the PR and shouldn't pull it.

**Comment by chulcha** [comment]
> 
> 
> Among other points, I sent a PR with a suggestion about this. Please see: [#7490 (comment)](https://github.com/opencart/opencart/pull/7490#issuecomment-505238993)
> 
> Much cleaner than it is currently:
> 
> `$data['results'] = $this->load->controller('common/pagination/results', $product_total);`
> 
> Daniel ignored the PR and shouldn't pull it.

WHERE $page $limit?
These variables may be different in controllers.

**Comment by diegospm** [comment]
The `$page` variable is captured in the Request object `$this->request->get ['page']`. If it is not found then it is set to 1. I stored this variable in the Provider object I sent in PR. So just call `$this->provider->page`. In short, the `$page` variable has become a kind of global variable that can be accessed from any controller.

The `$limit` variable usually is accessed from the Config object `$this->config->get ('config_limit_admin')`. In this case I passed it to the scope of the paging class and also left the second argument of the result method to pass the variable limit if different.

Please, see: https://github.com/opencart/opencart/blob/81901ac7e6eb47f73705cd901438f0528be2db7c/upload/admin/controller/common/pagination.php

And: https://github.com/opencart/opencart/blob/81901ac7e6eb47f73705cd901438f0528be2db7c/upload/system/library/provider.php

**Comment by diegospm** [comment]
The pagination was also cleaner: `$data['pagination'] = $this->load->controller('common/pagination', $product_total);`

**Comment by diegospm** [comment]
In addition, it also reduced the `$url` string repetition, as it currently is necessary to restart the string for each link variable. In the Provider object I reuse the current url and replace only the key needed for the new link.

Why? Reduces lines of code, makes it easier to read/maintain and also the developing the extensions for devs :)

**Comment by chulcha** [comment]
no, for еxample pagination for product review or any conroller(module)

**Comment by chulcha** [comment]
> 
> 
> In addition, it also reduced the `$url` string repetition, as it currently is necessary to restart the string for each link variable. In the Provider object I reuse the current url and replace only the key needed for the new link.
> 
> Why? Reduces lines of code, makes it easier to read/maintain and also the developing the extensions for devs :)

see url for sort or order

**Comment by diegospm** [comment]
> no, for еxample pagination for product review or any conroller(module)

give me an example?

**Comment by diegospm** [comment]
> > In addition, it also reduced the `$url` string repetition, as it currently is necessary to restart the string for each link variable. In the Provider object I reuse the current url and replace only the key needed for the new link.
> > Why? Reduces lines of code, makes it easier to read/maintain and also the developing the extensions for devs :)
> 
> see url for sort or order

See the working Provider object: https://github.com/diegospm/opencart/blob/refactoring/upload/admin/controller/catalog/product.php

sort and order:
`
$sorts = array('sort_name' => 'pd.name', 'sort_model' => 'p.model', 'sort_price' => 'p.price', 'sort_quantity' => 'p.quantity', 'sort_order' => 'p.sort_order');

foreach ($sorts as $key => $sort) {
 $data[$key] = $this->provider->link('catalog/product', array('sort' => $sort, 'order' => $this->provider->order));
}
`

**Comment by chulcha** [comment]
> 
> 
> > no, for еxample pagination for product review or any conroller(module)
> 
> give me an example?

product/product/review

**Comment by diegospm** [comment]
Originally, I sent the PR to solve a problem in the admin controllers. But could use it in the catalog, taking as an example the class you mentioned:

`$data['pagination'] = $this->load->controller('common/pagination', $product_total, 0, 'product/product/review', ['product_id' => $this->request->get['product_id']]);`

`$data['results'] = $this->load->controller('common/pagination/results', $product_total, 5);`

In this case, will need to make a slight change to the paging class:

`public function index(int $total, int $limit = 0, string $route = '', array $params = []) {`

`$provider_url = $this->provider->link($route, array_merge($params, ['page' => '{page}']));`

**Comment by diegospm** [comment]
Based on the PR I sent.
